### PR TITLE
Workaround sporadic failures of developer mode test

### PR DESCRIPTION
**Pull request by Martchus**
* Retry re-login in developer mode test
* Related ticket: https://progress.opensuse.org/issues/128807

### CODE REVIEW

**Comment by okurz** [comment]
s/sporatic/sporadic/.

Now *I* am interested to know what fail ratios and confidence you found:) I suggest you use https://github.com/okurz/scripts/blob/master/count_fail_ratio for that

File: t/33-developer_mode.t [hunk 1]
```diff
@@ -80,6 +80,23 @@ $ws = create_websocket_server(undef, 0, 0);
 $scheduler = create_scheduler;
 $livehandler = create_live_view_handler;
 
+# logs out and logs in again as the specified user; tries multiple times to workaround poo#128807
+my $max_login_attempts = $ENV{OPENQA_DEVEL_MODE_TEST_MAX_LOGIN_ATTEMPTS} // 10;
+sub relogin_as ($user) {
+    my $login_text = '';
+    my $expected_login_text = 'Logged in as ' . $user;
+    for (my $attempts = 1;; ++$attempts) {
```
Comment (Martchus) [inline t/33-developer_mode.t:88]: I actually need to check for `$max_login_attempts` here.

File: t/33-developer_mode.t [hunk 2]
```diff
@@ -80,6 +80,23 @@ $ws = create_websocket_server(undef, 0, 0);
...
+my $max_login_attempts = $ENV{OPENQA_DEVEL_MODE_TEST_MAX_LOGIN_ATTEMPTS} // 10;
+sub relogin_as ($user) {
+    my $login_text = '';
+    my $expected_login_text = 'Logged in as ' . $user;
+    for (my $attempts = 0; $attempts < $max_login_attempts; ++$attempts) {
+        if ($login_text ne 'Login') {
+            $driver->get('/logout');
+            $driver->element_text_is('#user-action a', 'Login', 'logged-out before logging in as ' . $user);
+        }
+        $driver->get('/login?user=' . $user);
```
Comment (okurz) [inline t/33-developer_mode.t:93]: that's unexpected. Why is not covered now?

Comment (Martchus) [reply t/33-developer_mode.t:93]: I also don't understand this. If it was not covered we would never execute `return pass` which would result in a test failure. I suspect the coverage tracking is broken.

**Comment by Martchus** [comment]
With the additional "Fix race condition …" commit which is about a different problem in the same test it seems pretty stable on my local machine:

```
## count_fail_ratio: Run: 35. Fails: 0. Fail ratio 0%
```

I'll keep this running for a little bit longer.

**Comment by Martchus** [comment]
```
## count_fail_ratio: Run: 97. Fails: 0. Fail ratio 0±0%. No fails, computed failure probability < 3.09%
```

Together with the previously mentioned runs this should be good enough.

**Comment by Martchus** [comment]
I cannot reproduce the uncovered line when generating a coverage report locally and it also makes no sense at all. I'll try marking the line as uncoverable. Maybe that helps with the report on codecov.

**Comment by Martchus** [comment]
Looks like `# uncoverable statement (must be bug in coverage tracking)` helped. (I first forced-pushed without it and the missing coverage persisted. So just re-triggering alone didn't help.)

**Comment by Martchus** [comment]
> Please remove it from the list of unstable tests as well now.

done

**Review by okurz — APPROVED** [review]

**Review by perlpunk — APPROVED** [review]

**Pull request merged**